libs/components/tabs/src/lib/modules/tabs/tabset-nav-button.component.ts

```diff
@@ -23,126 +20,174 @@ const buttonTypePrevious = 'previous';
   selector: 'sky-tabset-nav-button',
   templateUrl: './tabset-nav-button.component.html',
 })
-export class SkyTabsetNavButtonComponent implements AfterViewInit {
+export class SkyTabsetNavButtonComponent implements OnDestroy {
+  /**
+   * Specifies the tabset wizard component to associate with the nav button.
+   * @required
+   */
   @Input()
-  public tabset: SkyTabsetComponent;
+  public set tabset(value: SkyTabsetComponent) {
+    this.#_tabset = value;
 
-  @Input()
-  public buttonType: string;
+    if (this.#currentTabsetSub) {
+      this.#currentTabsetSub.unsubscribe();
+    }
+
+    if (value) {
+      if (this.#_buttonType === buttonTypeFinish) {
+        this.#_tabset.hasFinishButton = true;
+      }
+
+      this.#currentTabsetSub = this.#_tabset.activeChange
+        .pipe(distinctUntilChanged(), takeUntil(this.#ngUnsubscribe))
+        .subscribe((index: SkyTabIndex) => {
+          this.#activeSkyTabIndex = index;
+          this.#tabCount = this.#_tabset.tabs.length;
+          this.#updateTabToSelect();
+          this.#updateButtonVisibility();
+          this.#updateButtonProperties();
+        });
+    } else {
+      this.#logger.error(
+        'The SkyTabsetNavButtonComponent requires a reference to the SkyTabsetComponent it controls.'
+      );
+    }
+  }
 
+  /**
+   * Specifies the type of nav button to include.
+   * @required
+   */
   @Input()
-  public set buttonText(value: string) {
-    this._buttonText = value;
+  public get buttonType(): SkyTabsetNavButtonType {
+    return this.#_buttonType;
   }
 
-  public get buttonText(): string {
-    if (this._buttonText) {
-      return this._buttonText;
-    }
+  public set buttonType(value: SkyTabsetNavButtonType) {
```

Comment:
This is technically a breaking change... I hate it... but it is....

We took in any string before and so there is no guarantee that consumers aren't doing something dumb. We should probably try to handle random strings like we were previously with the `default` case in the `buttonText` setter. Happy to chat around this if needed.

Reply:
If we would even handle this with a union of your new type and `string` for now and then do a breaking change in 7 to remove that.

Reply:
I'm not sure I'm following. The logic has changed a bit, but they can still pass any string for button text. The template renders that if it's been passed, or the resource string if it hasn't. I think the diff for this just is comparing button text and type a little oddly

Reply:
ohhh nvm I see what you mean now 😬